Kernel/FileSystem/MountFile.cpp

```diff
@@ -0,0 +1,116 @@
+    : m_flags(flags)
+    , m_file_system_initializer(file_system_initializer)
+{
+    m_mount_specific_data.with_exclusive([&](auto& our_mount_specific_data) {
+        our_mount_specific_data = move(mount_specific_data);
+        memset(our_mount_specific_data->data(), 0, our_mount_specific_data->size());
+    });
+}
+
+MountFile::~MountFile() = default;
+
+ErrorOr<void> MountFile::ioctl(OpenFileDescription&, unsigned request, Userspace<void*> arg)
+{
+    return m_mount_specific_data.with_exclusive([&](auto& our_mount_specific_data) -> ErrorOr<void> {
+        switch (request) {
+        case MOUNT_IOCTL_SET_MOUNT_SPECIFIC_FLAG: {
+            auto user_mount_specific_data = static_ptr_cast<mount_specific_flag const*>(arg);
+            auto mount_specific_data = TRY(copy_typed_from_user(user_mount_specific_data));
+            if ((mount_specific_data.value_type == mount_specific_flag::ValueType::SignedInteger || mount_specific_data.value_type == mount_specific_flag::ValueType::UnsignedInteger) && mount_specific_data.value_length != 8)
+                return EDOM;
+            if (mount_specific_data.key_string_length > 64)
+                return ENAMETOOLONG;
+            if (mount_specific_data.value_type != mount_specific_flag::ValueType::None && mount_specific_data.value_length == 0)
+                return EINVAL;
+            if (mount_specific_data.value_type != mount_specific_flag::ValueType::None && mount_specific_data.value_addr == nullptr)
+                return EFAULT;
+            if (mount_specific_data.value_type != mount_specific_flag::ValueType::ASCIIString && mount_specific_data.value_length > 64)
+                return E2BIG;
+            if (mount_specific_data.value_type == mount_specific_flag::ValueType::ASCIIString && mount_specific_data.value_length > 1024)
+                return E2BIG;
+            // NOTE: We enforce that the passed argument will be either i64 or u64, so it will always be
+            // exactly 8 bytes. We do that to simplify handling of integers as well as to ensure ABI correctness
+            // in all possible cases.
+            auto key_string = TRY(try_copy_kstring_from_user({ reinterpret_cast<char const*>(mount_specific_data.key_string_addr), static_cast<size_t>(mount_specific_data.key_string_length) }));
+            switch (mount_specific_data.value_type) {
+            // NOTE: This is actually considered as simply boolean flag.
+            case mount_specific_flag::ValueType::None: {
+                VERIFY(m_file_system_initializer.handle_mount_boolean_flag);
+                Userspace<u64*> user_value_addr((FlatPtr)mount_specific_data.value_addr);
+                auto value_integer = TRY(copy_typed_from_user(user_value_addr));
+                if (value_integer != 0 && value_integer != 1)
+                    return EDOM;
+                bool value = (value_integer == 1) ? true : false;
```

Comment:
`ValueType::None` rubs me the wrong way if the value is communicated as a boolean. Maybe call it `ValueType::Boolean` instead?